modules/client/src/internal/client/decoding.ts

```diff
@@ -165,23 +165,8 @@ export class ClientDecoding extends ClientCore implements IClientDecoding {
     }
     const result = daoInterface.decodeFunctionData("setMetadata", data);
     const bytes = hexToBytes(result[0]);
-    const contentUriText = new TextDecoder().decode(bytes);
-    const contentUri = new URL(contentUriText);
-    if (contentUri.protocol !== "ipfs:") {
-      throw new UnsupportedProtocolError(contentUri.protocol);
-    }
-    const cid = contentUri.host;
-    const ipfsRegex =
-      /^Qm([1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})$/;
-    if (!ipfsRegex.test(cid)) {
-      throw new Error("The metadata URL defined on the DAO is invalid");
-    }
-    try {
-      const stringMetadata = await this.ipfs.fetchString(cid);
-      return JSON.parse(stringMetadata);
-    } catch {
-      throw new Error("Error reading data from IPFS");
-    }
+    const cid = new TextDecoder().decode(bytes);
+    return resolveIpfsCid(cid);
```

Comment:
We were returning the new metadata, whereas we are now returning the CiD, why is this?

Reply:
My bad I forgot about the `updateMetadataRawAction`